src/cloud_detection.c

```diff
@@ -0,0 +1,701 @@
+#include <stdbool.h>
+
+bool isCloud(float b, float r, float nir, float swir) {
```

Comment:
Where did this come from?

Also could you add the header and comments to this file/function?